sopel/coretasks.py

```diff
@@ -496,49 +501,88 @@ def track_modes(bot, trigger):
         "Y": module.OPER,
     }
 
-    # Parse modes before doing anything else
-    modes = []
-    sign = ''
+    # Process modes
+    sign = ""
+    param_idx = 0
```

Comment:
As mentioned below, I'd put that here, since it's not modified within the loop:

```suggestion
    param_idx = 0
    chanmodes = bot.isupport.CHANMODES
```